alpaca/broker/models/requests.py

```diff
@@ -454,3 +460,159 @@ def root_validator(cls, values: dict) -> dict:
+        street_address (Optional[str]): The street address of the bank, if and only if creating an international bank
+          account connection.
+    """
+
+    name: str
+    bank_code_type: IdentifierType
+    bank_code: str
+    account_number: str
+    country: Optional[str]
+    state_province: Optional[str]
+    postal_code: Optional[str]
+    city: Optional[str]
+    street_address: Optional[str]
+
+    @root_validator()
+    def root_validator(cls, values: dict) -> dict:
+        if "bank_code_type" not in values:
+            # Bank code type was not valid, so a ValueError will be thrown regardless.
+            return values
+
+        international_parameters = [
+            "country",
+            "state_province",
+            "postal_code",
+            "city",
+            "street_address",
+        ]
+
+        bank_code_type = values["bank_code_type"]
+        if bank_code_type == IdentifierType.ABA:
+            for international_param in international_parameters:
+                if (
+                    international_param in values
+                    and values[international_param] is not None
+                ):
+                    raise ValueError(
+                        f"You may only specify the {international_param} for international bank accounts."
+                    )
+        elif bank_code_type == IdentifierType.BIC:
+            for international_param in international_parameters:
+                if (
+                    international_param not in values
+                    or values[international_param] is None
+                ):
+                    raise ValueError(
+                        f"You must specify the {international_param} for international bank accounts."
+                    )
+
+        return values
+
+
+class CreateTransferRequest(NonEmptyRequest):
+    """
+    Attributes:
+        transfer_type (TransferType): Type of the transfer.
+        relationship_id (Optional[UUID]): ID of the relationship to use for the transfer, required for ACH transfers.
+        bank_id (Optional[UUID]): ID of the bank to use for the transfer, required for wire transfers.
+        amount (str): Amount of transfer, must be > 0. Any applicable fees will be deducted from this value.
+        direction (TransferDirection): Direction of the transfer.
+        timing (TransferTiming): Timing of the transfer.
+        additional_information (Optional[str]): Additional wire transfer details.
+        fee_payment_method (Optional[FeePaymentMethod]): Determines how any applicable fees will be paid. Default value
+          is invoice.
+    """
+
+    transfer_type: TransferType
+    relationship_id: Optional[UUID]
+    bank_id: Optional[UUID]
+    amount: str
+    direction: TransferDirection
+    timing: TransferTiming
+    additional_information: Optional[str]
+    fee_payment_method: Optional[FeePaymentMethod]
+
+    @root_validator()
+    def root_validator(cls, values: dict) -> dict:
```

Comment:
Hmm, wont reject on this as it can be fixed up later before 1.0 but we should probably split this apart into the different kinds of transfer requests with no Optional fields rather than having the complicated root validator :thinking:

Reply:
Refactored to two separate models, but produced some smelly code in the process (not too familiar with Pydantic). Let me know if there is a better implementation pattern!